corehq/apps/data_dictionary/views.py

```diff
@@ -107,3 +157,58 @@ def main_context(self):
     @memoized
     def section_url(self):
         return reverse(DataDictionaryView.urlname, args=[self.domain])
+
+
+class UploadDataDictionaryView(BaseProjectDataView):
+    section_name = _("Data Dictionary")
+    template_name = "data_dictionary/import_data_dict.html"
+    urlname = 'upload_data_dict'
+
+    @method_decorator(login_and_domain_required)
+    @use_jquery_ui
+    @toggles.DATA_DICTIONARY.required_decorator()
+    def dispatch(self, request, *args, **kwargs):
+        return super(UploadDataDictionaryView, self).dispatch(request, *args, **kwargs)
+
+    @property
+    def page_context(self):
+        main_context = super(UploadDataDictionaryView, self).main_context
+        main_context.update({
+            'bulk_upload': {
+                "download_url": reverse('export_data_dictionary', args=[self.domain]),
+                "adjective": _("data dictionary"),
+                "plural_noun": _("data dictionary"),
+            },
+        })
+        main_context.update({
+            'bulk_upload_form': get_bulk_upload_form(main_context),
+        })
+        return main_context
+
+    @method_decorator(atomic)
+    def post(self, request, *args, **kwargs):
+        bulk_file = self.request.FILES['bulk_upload_file']
+        filename = make_temp_file(bulk_file.read(), file_extention_from_filename(bulk_file.name))
+        with open_any_workbook(filename) as workbook:
+            for worksheet in workbook.worksheets:
+                case_type = worksheet.title
+                for row in itertools.islice(worksheet.iter_rows(), 1, None):
+                    name, group, data_type, description = [cell.value for cell in row[:4]]
+                    if name:
+                        prop = CaseProperty.get_or_create(
+                            name=name, case_type=case_type, domain=self.domain
+                        )
+                        if data_type:
+                            prop.data_type = data_type
+                        if description:
+                            prop.description = description
+                        if group:
+                            prop.group = group
+                        try:
+                            prop.full_clean()
+                        except ValidationError as e:
+                            messages.error(request, unicode(e))
+                            return self.get(request, *args, **kwargs)
```

Comment:
This seems like it will only show the first error, would be good to be able to show them all for each import.